src/main/scala/nl/vroste/zio/kinesis/client/fake/DynamicConsumerFake.scala

```diff
@@ -50,17 +48,15 @@ private[client] class DynamicConsumerFake(
     shards.flatMap {
       case (shardName, stream) =>
         ZStream.fromEffect {
-          ZIO.environment[R with Blocking].flatMap { env =>
-            CheckpointerFake.make(refCheckpointedList).map { checkpointer =>
-              (
-                shardName,
-                stream.zipWithIndex.mapM {
-                  case (byteBuffer, i) =>
-                    deserializer.deserialize(byteBuffer).flatMap(record(shardName, i, _)).provide(env)
-                },
-                checkpointer
-              )
-            }
+          CheckpointerFake.make(refCheckpointedList).map { checkpointer =>
+            (
+              shardName,
+              stream.zipWithIndex.mapM {
+                case (data, i) =>
+                  record(shardName, data, i)
+              },
```

Comment:
The deserializer parameter of `shardedStream` is not used here, which is a bit unexpected.